### PR TITLE
Rename file support in sql projects

### LINKED ISSUE #15312: Rename Files in Project
It would be nice to able to copy, paste, and rename files in a sql project.

### PR DESCRIPTION
**Pull request by kisantia**
This PR fixes #15312. This adds support for renaming files in a sql project. This will swap over to using the Move api in the DacFx projects when we move to use those apis.

Input boxes in treeviews aren't currently supported, which is why a quickpick is used for entering the new file name.
 
fileRename

error when a file with the same name at that location already exists:
image

### CODE REVIEW

File: extensions/sql-database-projects/src/controllers/projectController.ts [hunk 1]
```diff
@@ -839,6 +839,34 @@ export class ProjectsController {
...
+			});
+
+		if (!newFileName) {
+			return;
+		}
+
+		// TODO: swap this out and hookup to "Move" file/folder api
+		// rename the file
+		const newFilePath = path.join(path.dirname(file?.fsUri.fsPath!), `${newFileName}.sql`);
+		await fs.rename(file?.fsUri.fsPath!, newFilePath);
```
Comment (SakshiS-harma) [inline extensions/sql-database-projects/src/controllers/projectController.ts:863]: What will happen if a file with same name exists?

Comment (kisantia) [reply extensions/sql-database-projects/src/controllers/projectController.ts:863]: Added validation: 
image

File: extensions/sql-database-projects/package.json [hunk 2]
```diff
@@ -426,6 +435,11 @@
           "when": "view == dataworkspace.views.main && viewItem == databaseProject.itemType.folder || viewItem =~ /^databaseProject.itemType.file/ || viewItem == databaseProject.itemType.reference || viewItem == databaseProject.itemType.sqlcmdVariable",
           "group": "9_dbProjectsLast@2"
         },
+        {
+          "command": "sqlDatabaseProjects.rename",
+          "when": "view == dataworkspace.views.main && viewItem =~ /^databaseProject.itemType.file/",
```
Comment (Benjin) [inline extensions/sql-database-projects/package.json:440]: Not for folders?  Or maybe I've just forgotten how the conditional here works.

**Comment by Benjin** [comment]
Any plans to add a test for this?  or not until the new backend is stitched in?

File: extensions/sql-database-projects/package.json (hunk 2 above)
Comment (kisantia) [reply extensions/sql-database-projects/package.json:440]: only files for now. I checked the change I made in DacFx and it only supported moving files, so I'll have to go back and add support for moving folders.

**Comment by kisantia** [comment]
> Any plans to add a test for this? or not until the new backend is stitched in?

Yeah I was planning on adding tests when the new backend is swapped in, so it isn't a waste in case the test would need a complete rewrite. I can add one though if you think that'd be better

File: extensions/sql-database-projects/src/controllers/projectController.ts (hunk 1 above)
Comment (SakshiS-harma) [reply extensions/sql-database-projects/src/controllers/projectController.ts:863]: I don't see any code changes, any chance you might have forgotten to push the changes.

Comment (kisantia) [reply extensions/sql-database-projects/src/controllers/projectController.ts:866]: whoops pushed the changes now

**Review by SakshiS-harma — APPROVED** [review]

**Review by kburtram — APPROVED** [review]

**Pull request merged**